common/utils.go

```diff
@@ -56,3 +58,18 @@ func FileExists(path string) (bool, error) {
 
 	return false, err
 }
+
+// SleepWithContext is the same as time.Sleep(...) but with preemption if the context is
+// complete. Returns nil if the sleep completed, otherwise the context's error.
+func SleepWithContext(ctx context.Context, d time.Duration) error {
```

Comment:
This was moved from ethereum/block to make it available to other packages. This version now returns an error type so the caller knows if the sleep was interrupted.